### PR TITLE
`azurerm_security_center_setting` - `setting_name` supports `WDATP_EXCLUDE_LINUX_PUBLIC_PREVIEW` and `SENTINEL`

### LINKED ISSUE #18344: Support for bi-directional sync for Microsoft Defender for Cloud Sentinel data connector
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

It is possible to create a Microsoft Sentinal Data Connector for Microsoft Defender for cloud using the `azurerm_sentinel_data_connector_azure_security_center` resource.

Unfortunatelly, the bidirectional sync is not supported, which forces us to use a provisioner to enabled it. Adding support for this feature would be welcome.

### New or Affected Resource(s)/Data Source(s)

azurerm_sentinel_data_connector_azure_security_center

### Potential Terraform Configuration

```hcl
resource "azurerm_sentinel_data_connector_azure_security_center" "defender-cloud-connector" {
  name                       = "name"
  log_analytics_workspace_id = "xxx"
  subscription_id            = "yyy"
  bi_directional_sync        = true
}
```

### References

_No response_

### PR DESCRIPTION
**Pull request by magodo**
Fix #18344 

## Test

```shell
>  TF_ACC=1 go test -v -timeout=20h ./internal/services/securitycenter -run=TestAccSecurityCenterSetting_
=== RUN   TestAccSecurityCenterSetting_update
--- PASS: TestAccSecurityCenterSetting_update (81.13s)
=== RUN   TestAccSecurityCenterSetting_requiresImport
--- PASS: TestAccSecurityCenterSetting_requiresImport (28.55s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/securitycenter        109.697s
```

### CODE REVIEW

File: internal/services/securitycenter/security_center_setting_resource.go [hunk 1]
```diff
@@ -44,6 +43,8 @@ func resourceSecurityCenterSetting() *pluginsdk.Resource {
 				ValidateFunc: validation.StringInSlice([]string{
 					"MCAS",
 					"WDATP",
+					"WDATP_EXCLUDE_LINUX_PUBLIC_PREVIEW",
```
Comment (tombuildsstuff) [inline internal/services/securitycenter/security_center_setting_resource.go:46]: presumably this one will be removed in time?

**Review by tombuildsstuff — APPROVED** [review]
Left one comment inline but this otherwise LGTM 👍

File: internal/services/securitycenter/security_center_setting_resource.go (hunk 1 above)
Comment (tombuildsstuff) [reply internal/services/securitycenter/security_center_setting_resource.go]: I'm not suggesting we remove this necessarily, fwiw - more a question than anything?

Comment (magodo) [reply internal/services/securitycenter/security_center_setting_resource.go]: I'm not sure, but per the [CLI document](https://learn.microsoft.com/en-us/cli/azure/security/setting?view=azure-cli-latest#az-security-setting-list-examples), it is still supported. Also it shows up in my subscription when I list the settings. Based on this, shall I add it back?

**Review by katbyte — APPROVED** [review]

**Pull request merged**